### DISCUSSION TITLE
Different result when changing order of particles

### DISCUSSION THREAD

**Discussion by ariaradick** [FeynCalc/feyncalc#272] [Q&A]
Hi! I have a FeynRules-generated FeynArts model file that I'm using with FeynCalc that adds a charged scalar with the same charges as the left-handed leptons (and a neutral scalar w/ same charges as neutrinos), and I'm finding that I get different results for generating [A phi- > A phi-] vs [phi- A > phi- A],
```
diags = InsertFields[
   CreateTopologies[0, 2 -> 2], {V[1], S[4]} -> {V[1], S[4]}, 
   InsertionLevel -> {Classes}, Model -> "LFDM_LL_FA", 
   GenericModel -> "LFDM_LL_FA"];
amp[0] = 
 FCFAConvert[CreateFeynAmp[diags], IncomingMomenta -> {p1, p2}, 
  OutgoingMomenta -> {k1, k2}, UndoChiralSplittings -> True, 
  ChangeDimension -> 4, List -> False, SMP -> True, Contract -> True, 
  TransversePolarizationVectors -> {p1, k1}, 
  FinalSubstitutions -> {SMP["e"] -> Sqrt[4 \[Pi] \[Alpha]]}]

```
vs
```
diags9 = 
  InsertFields[
   CreateTopologies[0, 2 -> 2], {S[4], V[1]} -> {S[4], V[1]}, 
   InsertionLevel -> {Classes}, Model -> "LFDM_LL_FA", 
   GenericModel -> "LFDM_LL_FA"];
amp[9] = 
 FCFAConvert[CreateFeynAmp[diags9], IncomingMomenta -> {p2, p1}, 
  OutgoingMomenta -> {k2, k1}, UndoChiralSplittings -> True, 
  ChangeDimension -> 4, List -> False, SMP -> True, Contract -> True, 
  TransversePolarizationVectors -> {p1, k1}, 
  FinalSubstitutions -> {SMP["e"] -> Sqrt[4 \[Pi] \[Alpha]]}]
```

The first one gives
```
ampSquared[0] = (128 π^2 α^2 (-3 msca^8+4 msca^6 t+s^2 (-2 msca^2+s+t)^2+msca^4 ((-2 msca^2+s+t)^2+s^2)))/((msca^2-s)^2 (-msca^2+s+t)^2)
```

whereas the second one gives
```
ampSquared[9] = (32 π^2 α^2 (7 msca^8-2 msca^6 (14 s+3 t)+msca^4 (42 s^2+26 s t+7 t^2)-2 msca^2 s (14 s^2+17 s t+3 t^2)+7 s^2 (s+t)^2))/((msca^2-s)^2 (-msca^2+s+t)^2)
```

which differ by
```
ampSquared[9] - ampSquared[0] = 96 π^2 α^2
```

The correct answer should be
```
ampSqTrue = (256 π^2 α^2 (msca^8-msca^6 (4 s+t)+msca^4 (6 s^2+4 s t+t^2)-msca^2 s (4 s^2+5 s t+t^2)+s^2 (s+t)^2))/((msca^2-s)^2 (-msca^2+s+t)^2)
```

which also disagrees with _both_ results
```
ampSqTrue - ampSquared[0] = 128 π^2 α^2
ampSqTrue - ampSquared[9] = 32 π^2 α^2
```

This seems to indicate that the order in which I put the external particles matters (which it should not!), and that FeynCalc fails to give the correct result in this case.

However, since posting this I found that if I add a momentum substitution k1 -> p1+p2-k2 (where k1 is the momentum of the final state photon) before calculating the squared amplitude, the results agree both with the correct value and each other. Any insight as to what's happening here would be appreciated. Is this expected FeynCalc behavior or perhaps a bug?

**Comment by vsht** [comment]
Hi,

for anyone to have any ideas you should first explain the difference in the results.

The more generic questions one asks, the less likely it is to obtain helpful answers ;)

**Comment by ariaradick** [comment]
Thank you for your response, I have added the results of each calculation to the original post. Also since posting this I have found that doing a momentum substitution resolves the issue, however I wonder if this is really expected behavior?

**Reply by vsht** [reply]
Given that the model itself is correct and all other steps are also done right, I'd assume that you might be using
a wrong polarization formula for the vectors. Like just inserting $-g^{\mu \nu}$ instead of using auxiliary vectors
as one should.
